frontend/src/components/worksheets/BundleDetail/BundleFieldTable/BundleStateRow.jsx

```diff
@@ -14,36 +20,45 @@ import BundleFieldRow from './BundleFieldRow';
 class BundleStateTable extends React.Component {
     constructor(props) {
         super(props);
-        const states = this.getStates();
-        this.state = {
-            states,
-        };
+        const bundleType = this.props.bundle.bundle_type.value;
+        const states = this.getStates(bundleType);
+        this.state = { states };
     }
 
-    getStates() {
-        const bundleType = this.props.bundle.bundle_type.value;
+    getStates(bundleType) {
         if (bundleType === 'run') {
-            return ['created', 'staged', 'starting', 'preparing', 'running', 'finalizing', 'ready'];
+            return RUN_BUNDLE_STATES;
         }
         if (bundleType === 'dataset') {
-            return ['created', 'uploading', 'ready'];
+            return UPLOADED_BUNDLE_STATES;
         }
         if (bundleType === 'make') {
-            return ['created', 'making', 'ready'];
+            return MAKE_BUNDLE_STATES;
         }
         return [];
     }
 
+    showTime(time) {
+        return time && time !== '<none>' && time !== '0s';
```

Comment:
It seems like we should be able to access the raw value rather than the rendered one to avoid these rather brittle comparisons, so we could test whether `time == null` or something...

Reply:
will update the `renderFormat` helper to allow for this 👍